include/aikido/robot/ConcreteRobot.hpp

```diff
@@ -0,0 +1,262 @@
+#include "aikido/robot/util.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+#include "aikido/trajectory/Trajectory.hpp"
+
+namespace aikido {
+namespace robot {
+
+AIKIDO_DECLARE_POINTERS(ConcreteRobot)
+
+class ConcreteRobot : public Robot
+{
+public:
+  ConcreteRobot(
+      const std::string& name,
+      dart::dynamics::MetaSkeletonPtr robot,
+      statespace::dart::MetaSkeletonStateSpacePtr statespace,
+      bool simulation,
+      std::unique_ptr<aikido::common::RNG> rng,
+      std::shared_ptr<control::TrajectoryExecutor> trajectoryExecutor,
+      dart::collision::CollisionDetectorPtr collisionDetector,
+      dart::collision::CollisionGroupPtr collideWith,
+      std::shared_ptr<dart::collision::BodyNodeCollisionFilter>
+          selfCollisionFilter);
+
+  virtual ~ConcreteRobot() = default;
+
+  // Documentation inherited.
+  virtual std::unique_ptr<aikido::trajectory::Spline> smoothPath(
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const aikido::trajectory::Trajectory* path,
+      const constraint::TestablePtr& constraint) override;
+
+  // Documentation inherited.
+  virtual std::unique_ptr<aikido::trajectory::Spline> retimePath(
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const aikido::trajectory::Trajectory* path) override;
+
+  // Documentation inherited.
+  virtual void executeTrajectory(
+      const trajectory::TrajectoryPtr& trajectory) override;
+
+  // Documentation inherited.
+  virtual boost::optional<Eigen::VectorXd> getNamedConfiguration(
+      const std::string& name) const override;
+
+  // Documentation inherited.
+  virtual void setNamedConfigurations(
+      std::unordered_map<std::string, const Eigen::VectorXd>
+          namedConfigurations) override;
+
+  // Documentation inherited.
+  virtual std::string getName() const override;
+
+  // Documentation inherited.
+  virtual dart::dynamics::MetaSkeletonPtr getMetaSkeleton() override;
+
+  // Documentation inherited.
+  virtual aikido::statespace::dart::MetaSkeletonStateSpacePtr getStateSpace()
+      override;
+
+  // Documentation inherited.
+  virtual void setRoot(Robot* robot) override;
+
+  // Documentation inherited.
+  virtual void step(
+      const std::chrono::system_clock::time_point& timepoint) override;
+
+  // Documentation inherited.
+  virtual aikido::constraint::CollisionFreePtr getSelfCollisionConstraint(
+      const statespace::dart::MetaSkeletonStateSpacePtr& space,
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton) override;
+
+  // Documentation inherited.
+  virtual aikido::constraint::TestablePtr getFullCollisionConstraint(
+      const statespace::dart::MetaSkeletonStateSpacePtr& space,
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const constraint::CollisionFreePtr& collisionFree) override;
+
+  /// Plan the robot to a specific configuration. Restores the robot to its
+  /// initial configuration after planning.
```

Comment:
Will we still have these planning methods here after the `Planner` API has been added? If not, let's add TODOs to remove them.